### PR TITLE
Security: upgrade CSP headers

### PR DESCRIPTION
**Pull request by chohner**
- refactors into file + add unit tests
- add stricter directives
- add fallback for older browsers
- add references

We now full green checkmarks on CSP evaluator
image

todo:
- [x] document why we need `connect-src: *` in development?

### CODE REVIEW

File: app/services/security/cspHeader.server.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  [
+    `default-src 'self'`,
+    `script-src 'self' 'nonce-${cspNonce}' https://*.posthog.com 'strict-dynamic' https: 'unsafe-inline'`,
+    `style-src 'self' 'unsafe-inline'`,
+    `connect-src ${
+      config().ENVIRONMENT === "development"
+        ? "*"
+        : "'self' https://*.ingest.sentry.io https://*.posthog.com"
+    }`,
+    `img-src 'self' localhost:* https://a2j-rechtsantragstelle-infra-public-assets-bucket.obs.eu-de.otc.t-systems.com https://mermaid.ink data:`,
```
Comment (pgurusinga) [inline app/services/security/cspHeader.server.tsx:18]: I think we can conditionally set `localhost` for `dev` only — as far as I remember this is used for loading images when we start local Strapi (the images are referenced to the bucket)

File: tests/unit/services/security/cspHeader.test.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { cspHeader } from "~/services/security/cspHeader.server";
+
+describe("cspHeader", () => {
+  const header = cspHeader("");
+  it("should contain all OWASP recommended directives", () => {
```
Comment (pgurusinga) [inline tests/unit/services/security/cspHeader.test.ts:5]: what do you think if we add a test for checking the correct structure of the CSP directives? we might misconfigured e.g. missing semicolon or too many spaces.
```
default-src 'self'; script-src 'self'; style-src 'self' 'unsafe-inline'; connect-src 'self'; img-src 'self' localhost:*
```

**Review by aaschlote — APPROVED** [review]
Nicely done!

File: app/services/security/cspHeader.server.tsx (hunk 1 above)
Comment (chohner) [reply app/services/security/cspHeader.server.tsx:12]: I just went to the opposite direction with the `connect-src` directive. afaik including localhost on prod isn't harmful but the more I think about it, the more I agree we should strip all mentions of localhost...

File: tests/unit/services/security/cspHeader.test.ts (hunk 2 above)
Comment (chohner) [reply tests/unit/services/security/cspHeader.test.ts:5]: Actually, I'm not sure what to test against here. but a malformed CSP will immediately break the site so it should be very noticeable

**Pull request merged**